app/src/org/commcare/android/util/StorageUtils.java

```diff
@@ -80,11 +82,27 @@ public int compare(Integer lhs, Integer rhs) {
             }
         });
         
-        //The records should now be in order and we can pass to the next phase 
+        //The records should now be in order and we can pass to the next phase
         FormRecord[] records = new FormRecord[ids.size()];
         for(int i = 0 ; i < ids.size() ; ++i) {
             records[i] = storage.read(ids.elementAt(i).intValue());
         }
-        return records;
+
+        return filterByCurrentApp(records);
+    }
+
+    /**
+     * Filter the given list of forms down to only those that are part of the currently seated app
+     */
+    private static FormRecord[] filterByCurrentApp(FormRecord[] forms) {
```

Comment:
See comment above: I think our storage layer makes this the only possible option anyway, so this is likely not necessary.